lib/grape-swagger/entity/attribute_parser.rb

```diff
@@ -95,8 +95,11 @@ def document_data_type(documentation, data_type)
 
         type[:format] = documentation[:format] if documentation.key?(:format)
 
-        if (values = documentation[:values]) && values.is_a?(Array)
+        case (values = documentation[:values])
+        when Array
           type[:enum] = values
+        when Proc
+          type[:enum] = values.call
```

Comment:
Should we have an extra check that the `.call' returns an array? To avoid any surprises in the schema?

Reply:
@numbata Thanks for the review.
I've fixed it to check if the return value of Proc is an Array.